src/consumer/component/dependencies/dependency-resolver/dependencies-resolver.ts

```diff
@@ -328,32 +328,40 @@ export default class DependencyResolver {
       throw Error('traverseTreeForComponentId should get called only when rootDir is set');
     }
     const rootDirFullPath = path.join(this.consumerPath, this.componentMap.rootDir);
-    // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
-    if (this.tree[depFile].files && this.tree[depFile].files.length) {
-      // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
-      for (const file of this.tree[depFile].files) {
+    const files = this.tree[depFile].files || [];
+
+    if (files && !R.isEmpty(files)) {
+      for (const file of files) {
         const fullDepFile = path.resolve(rootDirFullPath, file.file);
         const depRelativeToConsumer = pathNormalizeToLinux(path.relative(this.consumerPath, fullDepFile));
         const componentId = this.consumer.bitMap.getComponentIdByPath(depRelativeToConsumer);
-        if (componentId) return componentId; // eslint-disable-line consistent-return
+        if (componentId) return componentId;
       }
     }
-    // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
-    if (this.tree[depFile].bits && this.tree[depFile].bits.length) {
-      // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
-      for (const bit of this.tree[depFile].bits) {
-        const componentId = this.consumer.getComponentIdFromNodeModulesPath(bit, this.component.bindingPrefix);
-        if (componentId) return componentId; // eslint-disable-line consistent-return
+    if (this.tree[depFile].bits && !R.isEmpty(this.tree[depFile].bits)) {
+      const bits = this.tree[depFile].bits || [];
+      for (const bit of bits) {
+        if (bit.componentId) {
+          return bit.componentId;
+        }
+        if (bit.fullPath) {
+          const componentId = this.consumer.getComponentIdFromNodeModulesPath(
+            bit.fullPath,
+            this.component.bindingPrefix
+          );
+          if (componentId) return componentId;
+        } else {
+          const componentId = packageNameToComponentId(this.consumer, bit.name, this.component.bindingPrefix);
+          return componentId;
+        }
```

Comment:
consider refactoring. it's starting to look like an arrow pattern.

Reply:
I agree it's not great at all. we are going to refactor this majorly anyway as part of the dependency-resolver task. it's not worth to refactor it now.